File: WOT-iOS/WOTData/WOTData/core/requests/TankList/WOTWEBRequestTanks.m
//
//  WOTWEBRequestTanks.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 6/5/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTWEBRequestTanks.h"
#import "WOTDataDefines.h"

@implementation WOTWEBRequestTanks

- (NSDictionary *)query {

    NSMutableDictionary *result = [[NSMutableDictionary alloc] init];
    result[WOTApiKeys.applicationId] = [NSString valueOrSpaceString:self.hostConfiguration.applicationID];
    result[WOTApiKeys.fields] = [NSString valueOrSpaceString:[self.args escapedValueForKey:WOTApiKeys.fields]];
    return result;
}

- (NSString *)path {
    
    return @"/wot/encyclopedia/tanks/";
}

@end

File: WOT-iOS/WOTData/WOTData/WebServices/requests/WOTWEBRequestTanks.swift
//
//  WOTWEBRequestTanks.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 1/14/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import Foundation

@objc
public class WOTWEBRequestTanks: WOTWEBRequest, WOTModelServiceProtocol {

    override public var method: String { return "POST" }

    @available(*, deprecated, message: "TO be refactored")
    @objc
    public static func modelClassName() -> String {
        return NSStringFromClass(Vehicles.self)
    }

    @available(*, deprecated, message: "TO be refactored")
    @objc
    public func instanceModelClass() -> AnyClass? {
        return NSClassFromString( type(of: self).modelClassName() )
    }

    override public var path: String {
        return "/wot/encyclopedia/tanks/"
    }
}
